File: java/resistor-color/src/main/java/ResistorColor.java
import java.util.Comparator;
import java.util.Map;
import java.util.HashMap;

class ResistorColor
{
    private Map<String, Integer> colors = new HashMap<String, Integer>() {{
        put("black", 0);
        put("brown", 1);
        put("red", 2);
        put("orange", 3);
        put("yellow", 4);
        put("green", 5);
        put("blue", 6);
        put("violet", 7);
        put("grey", 8);
        put("white", 9);
    }};

    int colorCode(String color)
    {
        return colors.get(color);
    }

    String[] colors()
    {
        return colors.keySet()
                .stream()
                .sorted(Comparator.comparing(key -> colors.get(key)))
                .toArray(String[]::new);
    }
}

File: c/resistor-color/src/resistor_color.c
#include "resistor_color.h"
#include <stdlib.h>

int colorCode(resistor_band_t resistor_band)
{
	return resistor_band;
}

resistor_band_t *colors()
{
	static resistor_band_t bands[] = {BLACK, BROWN, RED, ORANGE, YELLOW, GREEN, BLUE, VIOLET, GREY, WHITE};
	return bands;
}
